tools/exact-pcap-match.c

```diff
@@ -300,177 +231,97 @@ int main (int argc, char** argv)
-                    "Nansec TCP Dump" : "UNKNOWN";
-    magic_str = fhdr_inp.magic == TCPDUMP_MAGIC ? "TCP Dump" : magic_str_inp;
-    if (options.verbose)
-    {
-        printf ("Magic    0x%08x (%i) (%s)\n", fhdr_inp.magic, fhdr_inp.magic, magic_str);
-        printf ("Ver Maj  0x%04x (%i)\n", fhdr_inp.version_major,fhdr_inp.version_major);
-        printf ("Ver Min  0x%04x (%i)\n", fhdr_inp.version_minor, fhdr_inp.version_minor);
-        printf ("Thiszone 0x%08x (%i)\n", fhdr_inp.thiszone, fhdr_inp.thiszone);
-        printf ("SigFigs  0x%08x (%i)\n", fhdr_inp.sigfigs, fhdr_inp.sigfigs);
-        printf ("Snap Len 0x%08x (%i)\n", fhdr_inp.snaplen, fhdr_inp.snaplen);
-        printf ("Link typ 0x%08x (%i)\n", fhdr_inp.linktype, fhdr_inp.linktype);
-    }
+    ch_log_info("Loading input file %s...\n", options.input);
 
-    offset = 0;
     int64_t total_matched = 0;
     int64_t total_lost = 0;
     for (pkt_num = 0; !stop && pkt_num < options.max_inp + options.offset_ref;
-            pkt_num++)
-    {
-
+            pkt_num++){
         if (pkt_num && pkt_num % (1000 * 1000) == 0)
         {
-            ch_log_info("Processed %li,000,000 packets\n",
-                        pkt_num / 1000 / 1000);
+            ch_log_info("Processed %li,000,000 packets\n", pkt_num / 1000 / 1000);
         }
 
-        pcap_pkthdr_t pkt_hdr;
-        if (read_expect (fd_inp, &pkt_hdr, sizeof(pkt_hdr), &offset, true))
-        {
-            ch_log_error(
-                    "Could not read enough bytes from %s at offset %li, (%li required)\n",
-                    options.input, offset, sizeof(pkt_hdr));
-            ch_log_error("Ending now\n");
-            break;
-        }
+        pkt_info_t pkt_info;
+        pkt_info = pcap_buff_next_packet(&inp_buff);
+        pcap_pkthdr_t *inp_hdr = inp_buff.hdr;
 
-        if (pkt_hdr.caplen > fhdr_ref.snaplen || pkt_hdr.caplen < 64)
-        {
-            ch_log_error(
-                    "Error, packet length (%li) out of range [64,%li] %u offset=%li\n",
-                    pkt_hdr.caplen, fhdr_inp.snaplen, offset);
-        }
-
-        if (options.verbose && pkt_num >= options.offset_inp
-                && (pkt_hdr.len + sizeof(expcap_pktftr_t) < pkt_hdr.caplen))
-        {
-            ch_log_warn("Warning: packet len %li < capture len %li\n",
-                        pkt_hdr.len, pkt_hdr.caplen);
-        }
-
-        char pbuf[1024 * 64] = { 0 };
-        if (read_expect (fd_inp, &pbuf, pkt_hdr.caplen, &offset, true))
-        {
+        switch(pkt_info){
+        case PKT_EOF:
+            goto find_input_misses;
+        case PKT_OVER_SNAPLEN:
+            ch_log_fatal("Packet with index %d does not comply with snaplen: %d (data len is %d)\n", pkt_num, inp_buff.snaplen, inp_buff.hdr->len);
+            break;
+        case PKT_LEN_TRUNCATED:
+            ch_log_warn("Packet data length (%d) is < capture length (%d).\n", inp_hdr->len, inp_hdr->caplen);
+            break;
+        default:
```

Comment:
Don't love the default case here. IMHO better to have all cases explicitly listed.